### PR TITLE
Add initial Armada loader

### PR DESCRIPTION
**Pull request by mark-burnett**
This adds a basic armada loader.  To use it, you must put resources on the genesis host at `/etc/kubernetes/armada-loader/assets`, including the actual armada file named `promenade-armada.yaml`.

This uses the admin cert for now.

### CODE REVIEW

File: promenade/templates/genesis/etc/kubernetes/kubelet/manifests/armada-loader.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+              while true; do
+                sleep 60
+                if armada apply promenade-armada.yaml ; then
+                  break
+                fi
+              done
+            fi
+
+            rm -rf /etc/kubernetes/kubelet/manifests/armada-loader.yaml
+            sleep 10000
```
Comment (aric49) [inline promenade/templates/genesis/etc/kubernetes/kubelet/manifests/armada-loader.yaml]: Is there a reason why the Armada pod needs to sleep after deleting the manifest?

File: promenade/templates/genesis/etc/kubernetes/armada-loader/pki/armada-loader-key.pem [hunk 2]
```diff
@@ -0,0 +1 @@
+{{ config.get(kind='CertificateKey', name='admin')['data'] }}
```
Comment (aric49) [inline promenade/templates/genesis/etc/kubernetes/armada-loader/pki/armada-loader-key.pem:1]: I might be missing it, but where do these keys get generated from prior to the Jinja template getting created?

**Review by aric49 — COMMENTED** [review]
Looks good. Just a few clarifications, and I think everything is great.

File: promenade/templates/genesis/etc/kubernetes/kubelet/manifests/armada-loader.yaml (hunk 1 above)
Comment (mark-burnett) [reply promenade/templates/genesis/etc/kubernetes/kubelet/manifests/armada-loader.yaml]: Yeah, actually I don't want it to exit and get restarted.  I want to give kubelet time to kill it.

File: promenade/templates/genesis/etc/kubernetes/armada-loader/pki/armada-loader-key.pem (hunk 2 above)
Comment (mark-burnett) [reply promenade/templates/genesis/etc/kubernetes/armada-loader/pki/armada-loader-key.pem:1]: These are given as inputs to genesis and join scripts.  They are currently generated by the `promenade generate` command, but the long-term plan for their source is probably that will be provided externally.

**Review by gardlt — APPROVED** [review]
LGTM.

**Comment by gardlt** [comment]
What will the `promenade-armada.yaml` consist of.

**Pull request merged**